x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -677,42 +680,171 @@ public void testWrongInputParam() {
     }
 
     public void testPeriodAndDurationInRowAssignment() {
-        for (var unit : List.of("millisecond", "second", "minute", "hour", "day", "week", "month", "year")) {
+        for (var unit : TIME_DURATIONS) {
             assertEquals("1:5: cannot use [1 " + unit + "] directly in a row assignment", error("row a = 1 " + unit));
```

Comment:
I just realized that
```
row a = "2024-01-01"::datetime + "1 week"::date_period
```
actually works. Since this is is a weird enough edge case, maybe it's worth adding a csv test as well.